docs/source/resources/migration.md

```diff
@@ -0,0 +1,123 @@
+
+For example the following `APIDataset` in `catalog.yml` changes from the following in 0.18.x code:
+
+```yaml
+us_corn_yield_data:
+  type: api.APIDataSet
+  url: https://quickstats.nass.usda.gov
+  credentials: usda_credentials
+  params:
+    key: SOME_TOKEN
+    format: JSON
+```
+
+to this in 0.19.x:
+
+```yaml
+us_corn_yield_data:
+  type: api.APIDataSet
+  url: https://quickstats.nass.usda.gov
+  credentials: usda_credentials
+  load_args:
+    params:
+      key: SOME_TOKEN
+      format: JSON
+```
+
+#### The `layer` attribute in `catalog.yml` has moved
+From 0.19, the `layer` attribute at the top level has been  moved inside the `metadata` -> `kedro-viz` attribute. You need to update `catalog.yml` accordingly.
+
+The following `catalog.yml` entry changes from the following in 0.18.x code:
+
+```yaml
+companies:
+  type: pandas.CSVDataSet
+  filepath: data/01_raw/companies.csv
+  layer: raw
+```
+
+to this in 0.19.x:
+
+```yaml
+companies:
+  type: pandas.CSVDataSet
+  filepath: data/01_raw/companies.csv
+  metadata:
+    kedro-viz:
+      layer: raw
+```
+
+#### Dataset renaming
+In 0.19.0 we renamed dataset and error classes to follow the [Kedro lexicon](https://github.com/kedro-org/kedro/wiki/Kedro-documentation-style-guide).
+
+* Dataset classes ending with `DataSet` are replaced by classes that end with `Dataset`.
+* Error classes starting with `DataSet` are replaced by classes that start with `Dataset`.
+
+Note that all of the classes below are also importable from `kedro.io`; only the module where they are defined is listed as the location.
+
+| Type                        | Removed Alias               | Location                       |
+| --------------------------- | --------------------------- | ------------------------------ |
+| `AbstractDataset`           | `AbstractDataSet`           | `kedro.io.core`                |
+| `AbstractVersionedDataset`  | `AbstractVersionedDataSet`  | `kedro.io.core`                |
+| `CachedDataset`             | `CachedDataSet`             | `kedro.io.cached_dataset`      |
+| `LambdaDataset`             | `LambdaDataSet`             | `kedro.io.lambda_dataset`      |
+| `MemoryDataset`             | `MemoryDataSet`             | `kedro.io.memory_dataset`      |
+| `DatasetError`              | `DataSetError`              | `kedro.io.core`                |
+| `DatasetAlreadyExistsError` | `DataSetAlreadyExistsError` | `kedro.io.core`                |
+| `DatasetNotFoundError`      | `DataSetNotFoundError`      | `kedro.io.core`                |
+
+
+#### All other dataset classes are removed from the core Kedro repository (`kedro.extras.datasets`)
+You now need to install and import datasets from the [`kedro-datasets`](https://github.com/kedro-org/kedro-plugins/tree/main/kedro-datasets) package instead.
+
```

Comment:
### Configuration environment need to be declared in `settings.py` explicitly if you have custom arguments
For example, if you used `CONFIG_LOADER_ARGS`  in `settings.py` to read spark's configs, you need to add `base_env` and `default_run_env` explicitly.

Pre 0.19
```
CONFIG_LOADER_ARGS = {
#       "base_env": "base",
#       "default_run_env": "local",
    "config_patterns": {
        "spark": ["spark*", "spark*/**"],
    }
}
```
In 0.19.x:
```
CONFIG_LOADER_ARGS = {
      "base_env": "base",  #
      "default_run_env": "local",
          "config_patterns": {
              "spark": ["spark*", "spark*/**"],
          }
}
```

If you didn't use `CONFIG_LOADER_ARGS`, this is not needed because Kedro would set it by default.